src/main/java/woohakdong/server/api/service/admin/auth/AdminAuthService.java

```diff
@@ -0,0 +1,121 @@
+import woohakdong.server.domain.refreshToken.RefreshTokenRepository;
+
+import java.util.Date;
+
+import static woohakdong.server.common.exception.CustomErrorInfo.*;
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class AdminAuthService {
+
+    private final MemberRepository memberRepository;
+    private final RefreshTokenRepository refreshTokenRepository;
+    private final JWTUtil jwtUtil;
+    private final BCryptPasswordEncoder passwordEncoder;
+
+    @Transactional
+    public LoginResponse login(AdminLoginRequest loginRequest) {
+        Member admin = memberRepository.findByMemberProvideId(loginRequest.username())
+                .orElseThrow(() -> new CustomException(ADMIN_MEMBER_ID_NOT_FOUND));
+
+        if (!passwordEncoder.matches(loginRequest.password(), admin.getMemberPassword())) {
+            throw new CustomException(INVALID_ADMIN_PASSWORD);
+        }
+
+        String access = jwtUtil.createJwt("access", admin.getMemberProvideId(), admin.getMemberRole(), 600000L);
+        String refresh = jwtUtil.createJwt("refresh", admin.getMemberProvideId(), admin.getMemberRole(), 86400000L);
+
+        addRefreshEntity(admin.getMemberProvideId(), refresh, 86400000L);
+
+        return LoginResponse.builder()
+                .accessToken(access)
+                .refreshToken(refresh)
+                .build();
+    }
+
+    @Transactional
+    public void createAdmin(AdminJoinRequest joinRequest) {
+        if (memberRepository.findByMemberProvideId(joinRequest.username()).isPresent()) {
+            throw new CustomException(ADMIN_USERNAME_IS_ALREADY_USED);
+        }
+
+        String encryptedPassword = passwordEncoder.encode("1234");
+
+        Member admin = Member.builder()
+                .memberProvideId(joinRequest.username())
+                .memberName(joinRequest.name())
+                .memberRole("ADMIN_ROLE")
+                .memberEmail(joinRequest.email())
+                .memberPassword(encryptedPassword)
+                .build();
+
+        memberRepository.save(admin);
+    }
+
+    @Transactional
+    public void updateAdmin(AdminInfoUpdateRequest updateRequest) {
+        Member admin = getMemberFromJwtInformation();
+
+        if (memberRepository.findByMemberProvideId(updateRequest.username()).isPresent()) {
+            throw new CustomException(ADMIN_USERNAME_IS_ALREADY_USED);
+        }
+
+        String encryptedPassword = passwordEncoder.encode("1234");
+
+        admin.adminUpdate(updateRequest.username(), updateRequest.name(), updateRequest.email(), encryptedPassword);
+
+        memberRepository.save(admin);
+    }
+
+    public AdminInfoResponse getAdminInfo() {
+        Member admin = getMemberFromJwtInformation();
+
+        return AdminInfoResponse.builder()
+                .username(admin.getMemberProvideId())
+                .name(admin.getMemberName())
+                .email(admin.getMemberEmail())
+                .build();
```

Comment:
아래처럼 팩토리 메서드 패턴으로 빌더를 숨겨보는건 어떨까요?

```suggestion
        return AdminInfoResponse.from(admin);
```

Reply:
네네 알겠습니다.